### ISSUE TITLE
error handling raw string literals

### ISSUE THREAD

**Issue by guwirth** [SonarOpenCommunity/sonar-cxx#944]
There is a error handling / detecting raw string literals. Parser is not always able to detect string token correctly. In sample below end of string is not detected:
```C++
const std::regex RegexEscape(R"([.^$|()\[\]{}*+?\\])");
```

**Comment by guwirth** [comment]
fixed with #945
